ibis/backends/sqlite/compiler.py

```diff
@@ -1,36 +1,515 @@
+from ibis.backends.base.sqlglot.datatypes import SQLiteType
+from ibis.backends.base.sqlglot.rewrites import (
+    rewrite_first_to_first_value,
+    rewrite_last_to_last_value,
+)
+from ibis.common.temporal import DateUnit, IntervalUnit
 from ibis.expr.rewrites import rewrite_sample
 
+SQLite.Generator.TYPE_MAPPING |= {
+    sge.DataType.Type.BOOLEAN: "BOOLEAN",
+}
+
+
+@public
+class SQLiteCompiler(SQLGlotCompiler):
+    __slots__ = ()
+
+    dialect = "sqlite"
+    quoted = True
+    type_mapper = SQLiteType
+    rewrites = SQLGlotCompiler.rewrites + (
+        rewrite_sample,
+        rewrite_first_to_first_value,
+        rewrite_last_to_last_value,
+    )
+
+    NAN = sge.NULL
+    POS_INF = sge.Literal.number("1e999")
+    NEG_INF = sge.Literal.number("-1e999")
+
+    def _aggregate(self, funcname: str, *args, where):
+        expr = self.f[funcname](*args)
+        if where is not None:
+            return sge.Filter(this=expr, expression=sge.Where(this=where))
+        return expr
+
+    @singledispatchmethod
+    def visit_node(self, op, **kw):
+        return super().visit_node(op, **kw)
+
+    @visit_node.register(ops.Levenshtein)
+    @visit_node.register(ops.RegexSplit)
+    @visit_node.register(ops.StringSplit)
+    @visit_node.register(ops.IsNan)
+    @visit_node.register(ops.IsInf)
+    @visit_node.register(ops.Covariance)
+    @visit_node.register(ops.Correlation)
+    @visit_node.register(ops.Quantile)
+    @visit_node.register(ops.MultiQuantile)
+    @visit_node.register(ops.Median)
+    @visit_node.register(ops.ApproxMedian)
+    @visit_node.register(ops.ArrayColumn)
+    @visit_node.register(ops.ArrayConcat)
+    @visit_node.register(ops.ArrayStringJoin)
+    @visit_node.register(ops.ArrayCollect)
+    @visit_node.register(ops.ArrayContains)
+    @visit_node.register(ops.ArrayFlatten)
+    @visit_node.register(ops.ArrayLength)
+    @visit_node.register(ops.ArraySort)
+    @visit_node.register(ops.ArrayStringJoin)
+    @visit_node.register(ops.CountDistinctStar)
+    @visit_node.register(ops.IntervalBinary)
+    @visit_node.register(ops.IntervalAdd)
+    @visit_node.register(ops.IntervalSubtract)
+    @visit_node.register(ops.IntervalMultiply)
+    @visit_node.register(ops.IntervalFloorDivide)
+    @visit_node.register(ops.IntervalFromInteger)
+    @visit_node.register(ops.TimestampBucket)
+    @visit_node.register(ops.TimestampAdd)
+    @visit_node.register(ops.TimestampSub)
+    @visit_node.register(ops.TimestampDiff)
+    @visit_node.register(ops.StringToTimestamp)
+    @visit_node.register(ops.TimeDelta)
+    @visit_node.register(ops.DateDelta)
+    @visit_node.register(ops.TimestampDelta)
+    @visit_node.register(ops.TryCast)
+    def visit_Undefined(self, op, **kwargs):
+        return super().visit_Undefined(op, **kwargs)
+
+    @visit_node.register(ops.Cast)
+    def visit_Cast(self, op, *, arg, to) -> sge.Cast:
+        if to.is_timestamp():
+            if to.timezone not in (None, "UTC"):
```

Comment:
Unrelated, but this reminds me that we should store the timezone in a normalized form in `dt.Timestamp` rather than an arbitrary string.